### PR TITLE
Add Ability to Load Existing Calibration into Engineering Diffraction UI

### LINKED ISSUE #26848: Load an existing calibration into ISIS Engineering GUI
As a user I want the ability to enter a vanadium run number and calibration sample run number and load an existing calibration if one exists.

Add this functionality to the calibration tab. It would be good to think about reducing the duplication of the inputs and maybe reusing the `Vanadium #` and `Calibration Sample #` boxes and just having 2 buttons for load/create new? Have think about the options first and discuss them before implementation.

### PR DESCRIPTION
**Pull request by cailafinn**
**Description of work.**
Adds the ability to load a GSAS parameter file into the engineering diffraction interface. This sets the current calibration of the interface internally and also sets up the interface to redo the creation of a calibration if required.

**To test:**
1. Test in Mantid plot & workbench
2. Open the interface via Interfaces -> Diffraction -> Engineering Diffraction 2
    a. You may have to add `Diffraction/Engineering_DIffraction_2.py` to the mantid.properties file. 
3. Generate a new calibration (V: 307521 Cal: 305738) if you do not have one already.
4. Click the load radio button and browse to the created file (`~/Engineering_Mantid/Calibration/something.prm`)
5. Check that the right values were loaded into the GUI, you can test this by clicking the create new radio button and running the calibration again. 

Fixes #26848 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by DannyHindson — COMMENTED** [review]
I've got a few general comments about the new "load existing" feature:

1) The Vanadium # and  Calibration # that are read out of the .prm file are loaded into two edit boxes that sit under the "Create New Calibration" radio button. These two edit boxes appear to be only relevant to the action of creating a new calibration and so the user isn't given a v strong hint that these are now the "current" run numbers. I wonder if this could be clarified by moving the "Create New"\"Load Existing" radio buttons so they're next to each other

2) Should we be loading\creating the correction workspaces (engggui_vanadium_curves, engggui_vanadium_integration) when performing the "load existing calibration" action? This would mean the create new and load existing actions take the user to a similar place in terms of the workspaces that are present

3) In the issue linked to this PR it says "As a user I want the ability to enter a vanadium run number and calibration sample run number and load an existing calibration if one exists". This pull request instead loads the run numbers from a .prm file. Is the ability to load a calibration based on the user keying in vanadium\calibration sample run numbers still required?

Since I'm a bit new to this area of work perhaps on these points we could also get an opinion from @martyngigg ?

Also I noticed a couple of other things which aren't specifically related to this PR so can perhaps be dealt with as separate issues:

1) the EnggCalibrate algorithm has a parameter called DetectorPositions that (I think) is an input parameter and yet it is displayed under a "Outputs" heading in the UI
2) when the Engineering UI calls the EnggCalibrate algorithm it passes in the same workspace name for the FittedPeaks and OutputParametersTableName output parameters (eg engggui_calibration_bank_1). I think this means you don't get both outputs and instead one of them overwrites the other

File: scripts/Engineering/gui/engineering_diffraction/tabs/calibration/model.py [hunk 1]
```diff
@@ -193,34 +205,59 @@ def create_output_files(self, calibration_dir, difc, tzero, ceria_path, vanadium
...
+                                                   bank="all")
         # Both Banks
         file_path = calibration_dir + filename
         write_ENGINX_GSAS_iparam_file(file_path,
                                       difc,
                                       tzero,
-                                      ceria_run=ceria_no,
-                                      vanadium_run=vanadium_no)
+                                      ceria_run=ceria_path,
+                                      vanadium_run=vanadium_path)
```
[the remark this answers is not in the document]
Comment (ConorMFinn) [reply scripts/Engineering/gui/engineering_diffraction/tabs/calibration/model.py]: The old GUI uses the path to the files in the `.prm` file. I had the same thought early on, and replaced it with just the run number. It does make it easier to load the data back in if the path is saved, as the GUI spends less time searching the archive.

**Comment by martyngigg** [comment]
> I've got a few general comments about the new "load existing" feature:
> 
>     1. The Vanadium # and  Calibration # that are read out of the .prm file are loaded into two edit boxes that sit under the "Create New Calibration" radio button. These two edit boxes appear to be only relevant to the action of creating a new calibration and so the user isn't given a v strong hint that these are now the "current" run numbers. I wonder if this could be clarified by moving the "Create New""Load Existing" radio buttons so they're next to each other
> 
>     2. Should we be loading\creating the correction workspaces (engggui_vanadium_curves, engggui_vanadium_integration) when performing the "load existing calibration" action? This would mean the create new and load existing actions take the user to a similar place in terms of the workspaces that are present
> 
>     3. In the issue linked to this PR it says "As a user I want the ability to enter a vanadium run number and calibration sample run number and load an existing calibration if one exists". This pull request instead loads the run numbers from a .prm file. Is the ability to load a calibration based on the user keying in vanadium\calibration sample run numbers still required?
> 
> 
> Since I'm a bit new to this area of work perhaps on these points we could also get an opinion from @martyngigg ?
> 

1. This seems sensible to me.
2. My thought would be to load them as I agree with Danny that having the user in a similar place whether they create/load seems sensible.
3. @ConorMFinn Are we only loading the numbers because that's one the old one did? If the action is called "load" then it feels like it should do some actual loading or you can't tell if those numbers are going to be used to create a new one...

**Comment by DannyHindson** [comment]
Looks good thanks. I will approve once the Jenkins failures are cleared - just in case any of these turn out to be real.

The change that stops the Focus tab generating the engggui_vanadium_curves and engggui_vanadium_integration workspaces if it doesn't find them is good I think because it separates out what the Calibration\Focus tabs do nicely (I think previously either tab could have created these two workspaces depending on the actions the user selected).

Couple of comments:
- I note you've not moved the "load existing" radio button in the end. I still think the way the "load existing" action populates the vanadium #\calibration sample # edit boxes is a bit awkward, given that these edit boxes appear to be strongly linked to the "create new" action. However we can revisit this again in future - and we may be better placed to ponder this once the remaining tabs have taken shape a bit more
- the pending_calibration and current_calibration variables (in calibration\presenter.py) and the logic relating to them don't appear to me to be needed any more. Is that right? Again can pick up on a future PR if necessary

**Review by DannyHindson — APPROVED** [review]
Retested in workbench and MantidPlot following final commit relating to using the base class setEnabled. Disabling\enabling of edit boxes based on choice of load new\load existing is still fine

**Pull request merged**